packages/components/src/range-control/index.js

```diff
@@ -64,7 +60,7 @@ function RangeControl( {
 		// The input is valid, reset the local state property used to temporaly save the value,
 		// and call onChange with the new value as a number.
 		resetCurrentInput();
-		onChange( newNumericValue );
+		onChange( parseFloat( newValue ) );
 	};
 	const initialSliderValue = isFinite( value ) ?
```

Comment:
Aside: Should this be testing `currentInputValue` ?

Reply:
Yes while checking the code it seems currentInputValue should be used. I updated the code.